### PR TITLE
Fix CI

### LINKED ISSUE #33: Broken CI/CD
> Since June 15th, 2021, the building on travis-ci.org is ceased. Please use travis-ci.com from now on.

Automatic tests and Galaxy imports no longer happen.

### PR DESCRIPTION
**Pull request by cityofships**
Closes #33

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          - stable-2.11
+          - devel
+    steps:
+
+      - uses: actions/checkout@v2
+
+      - name: Installing dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install git+https://github.com/ansible/ansible.git@${{ matrix.ansible }}
```
Comment (markgoddard) [inline .github/workflows/pull_request.yml]: `pip install ansible==2.x.*`?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          pip install ansible-lint
+          ansible-galaxy collection install pulp.squeezer
```
Comment (markgoddard) [inline .github/workflows/pull_request.yml]: Not necessary?

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (cityofships) [reply .github/workflows/pull_request.yml]: packages renamed upstream to `ansible-base 2.10.*`, `ansible-core 2.11.*`
we could go with non-obvious mapping of ansible<4.0.0 to ansible-base 2.10.* and ansible>=4.0.0 to ansible-core 2.11.*

Comment (markgoddard) [reply .github/workflows/pull_request.yml]: You can still pip install ansible, it comes with batteries

**Comment by markgoddard** [comment]
Should I see actions run against this PR?

**Comment by cityofships** [comment]
> Should I see actions run against this PR?

Hopefully here: https://github.com/cityofships/ansible-role-os-networks/runs/2971574890?check_suite_focus=true

**Comment by cityofships** [comment]
I mean - no, here they won't run yet as the PR just initialises them; that run from my fork is the same version of the workflows submitted here.

So I suppose if we wanted to test the new jobs we could merge a noop workflow to initialise actions, then this would run the jobs?

**Comment by cityofships** [comment]
Not necessarily - it's just forks that have different behaviour https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by markgoddard** [comment]
Confused myself - I edited your comment rather than replying to it.

"So I suppose if we wanted to test the new jobs we could merge a noop workflow to initialise actions, then this would run the jobs?"

**Comment by markgoddard** [comment]
It may require approval to run jobs, but currently we don't have that option.

**Comment by cityofships** [comment]
'ansible<3' and 'ansible<4' are not working as intended, both installing ansible-base-2.10.11

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+name: Publish
+
+# Controls when the action will run.
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
```
Comment (markgoddard) [inline .github/workflows/publish.yml]: Tags include a v prefix

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Review by markgoddard — APPROVED** [review]

**Pull request merged**